pkg/util/util_test.go

```diff
@@ -276,3 +276,52 @@ func (s *tryUntilSuccSuite) TestShouldStopOnSuccess(c *C) {
 	}
 	c.Assert(callCount, Equals, 3)
 }
+
+type retryCtxSuite struct{}
+
+var _ = Suite(&retryCtxSuite{})
+
+func (s *retryCtxSuite) TestOnlyRetrySpecifiedTimes(c *C) {
+	ctx := context.Background()
+	var callCount int
+	err := RetryContext(ctx, 2, time.Microsecond, 3, func(ictx context.Context) error {
+		callCount++
+		return errors.New("Fail")
+	})
+	c.Assert(err, ErrorMatches, "Fail")
+	c.Assert(callCount, Equals, 2)
+}
+
+func (s *retryCtxSuite) TestRetryUntilTimeout(c *C) {
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
+	defer cancel()
+	var callCount int
+	err := RetryContext(ctx, 10, time.Millisecond, 2, func(ictx context.Context) error {
+		callCount++
+		return errors.New("Fail")
+	})
+	c.Assert(err, ErrorMatches, "Fail")
+	c.Assert(callCount, Less, 10)
+
+	callCount = 0
+	err = RetryContext(ctx, 10, time.Millisecond, 2, func(ictx context.Context) error {
+		<-ictx.Done()
+		return errors.New("Canceled")
+	})
+	c.Assert(err, ErrorMatches, "Canceled")
+	c.Assert(callCount, Equals, 0)
```

Comment:
What's the purpose of this check? `callCount` is never modified inside lines 307-310.

Reply:
Fixed.